crates/readme.md

```diff
@@ -0,0 +1,44 @@
+# Structure
+
+Below we briefly describe the purpose of each crate in this repository.
```

Comment:
While we're at it, we should likely add a readme to each crate individually as well.

Reply:
Yeah this can be done in another PR by the respective owners of the larger crates. Frontend for example is owned by you and evaluator/SSA by Guillaume.